lib/rubygems/basic_specification.rb

```diff
@@ -71,8 +71,11 @@ def contains_requirable_file?(file)
     elsif missing_extensions?
       @ignored = true
 
-      warn "Ignoring #{full_name} because its extensions are not built. " +
-        "Try: gem pristine #{name} --version #{version}"
+      if RUBY_ENGINE == platform || Gem::Platform.local === platform
```

Comment:
This `RUBY_ENGINE == platform` check doesn't seem right, and causes a failure when running the tests on TruffleRuby.
It should be `Gem::Platform::RUBY == platform`, right?
(as in the RUBY platform, the universal platform independent of the local platform)

cc @bjfish

Reply:
I suggested to replace this condition with `Gem::Platform.local === platform` in https://github.com/rubygems/rubygems/pull/2672#discussion_r263358460, but I think @hsbt misunderstood my suggestion and added the condition instead.

Reply:
So it should be just `if Gem::Platform.local === platform` then?
Will a not-precompiled locally installed and compiled gem report a non-RUBY platform?

Reply:
https://github.com/rubygems/rubygems/pull/3495